### PR TITLE
Add a block-based convenience initializer

### PR DESCRIPTION
**Pull request by irace**
Add a convenience initializer to avoid having to always call the full designated initializer: https://github.com/tumblr/XExtensionItem/issues/1

### CODE REVIEW

**Comment by mattbischoff** [comment]
>  continuous-integration/travis-ci/pr — The Travis CI build failed

This looks like a real failure from the log.

**Comment by irace** [comment]
Man, isn’t continuous integration great?!

File: XExtensionItem/XExtensionItemParameters.m [hunk 1]
```diff
@@ -13,6 +13,24 @@ @implementation XExtensionItemParameters
 
 #pragma mark - Initialization
 
+- (instancetype)initWithBlock:(void (^)(XExtensionItemMutableParameters *))initializationBlock {
+    NSParameterAssert(initializationBlock);
+    
+    XExtensionItemMutableParameters *parameters = [[XExtensionItemMutableParameters alloc] init];
+    
+    initializationBlock(parameters);
```
Comment (mattbischoff) [inline XExtensionItem/XExtensionItemParameters.m]: Even though we parameter assert, I still think we should nil check this so we don’t crash the app that includes XExtensionItem in the case of developer error.

File: Example/XExtensionItemExample/ViewController.m [hunk 2]
```diff
@@ -40,6 +40,14 @@ - (void)action {
         
         [mutableParameters copy];
     });
+//
```
Comment (mattbischoff) [inline Example/XExtensionItemExample/ViewController.m]: Adding this for illustration?

Comment (irace) [inline Example/XExtensionItemExample/ViewController.m]: Bleh, naw, accidental (:musical_note:)

**Pull request merged**